fix: make icons inside IconButtons flex to fit button container

okry opened the pull request:
- Icon SVGs now inherit height and width of wrapping div element.
- IconButtons make SVG div wrappers expand/contract to fit the button container wrapping them
- Had to remove padding from IconButtons since it was causing the interior icons to be offset by a significant amount

New hover indicator:
<img width="45" alt="Screen Shot 2021-01-11 at 10 48 13 AM" src="https://user-images.githubusercontent.com/25124347/104203757-839d8500-53fa-11eb-9797-ed0e6a6f3071.png">

New focus indicator:
<img width="114" alt="Screen Shot 2021-01-11 at 10 47 45 AM" src="https://user-images.githubusercontent.com/25124347/104203709-74b6d280-53fa-11eb-9d90-3e9339b2d5b6.png">


These indicators were taken from what we were using for the vanilla buttons and were tested to work with stretched buttons.

scurker reviewed (changes requested):
The outlines for the focus/hover states are now offset. 

![react cauldron icon in a hover state with an offset hover border](https://user-images.githubusercontent.com/1062039/103950689-ef35d880-5102-11eb-9436-50be736f79c0.png)

![react cauldron icon in a focus state with an offset focus border](https://user-images.githubusercontent.com/1062039/103950740-0aa0e380-5103-11eb-96d8-c6ff73ec533f.png)

I think one of the things that might make this difficult, is the specificity of the hover states depend on a known height/width of the button.

schne324 commented:
it looks like that focus ring issue already exists: https://cauldron.netlify.app/components/IconButton but I agree we should fix it

scurker commented:
> it looks like that focus ring issue already exists: https://cauldron.netlify.app/components/IconButton but I agree we should fix it

???

This is what I see on https://cauldron.netlify.app/components/IconButton:
![image](https://user-images.githubusercontent.com/1062039/104076872-09dc8000-51dd-11eb-9a22-16386d48737c.png)

There's a 1px spacing around the button icon, which doesn't exist in this PR. 

For comparison:

![image](https://user-images.githubusercontent.com/1062039/104076928-427c5980-51dd-11eb-9941-1655080104ae.png)

okry commented:
I've updated the indicators to reflect Aaron's designs by taking from what we were using for the other buttons. Updated icons are in the main description.

scurker reviewed (changes requested):
One more small minor change: Add `pointer-events: none` to the `.IconButton:before` pseudo element. We don't want the pseudo element to be triggering click events or activating when it's not supposed to. `pointer-events: none` should prevent both of these behaviors.

stephenmathieson commented on packages/styles/icon-button.css:
```diff
@@ -1,42 +1,42 @@
   color: var(--icon-button-icon-color);
   margin: 2px;
+  padding: 0;
+  --button-hover-outline-color: var(--top-bar-text-color);
```
This looks like a typo. I don't think `--button-hover-outline-color` is a valid CSS property.

okry replied:
Wasn't a typo, it's a custom key that we were using for the hover outline. I essentially stole this methodology from the vanilla button css. Since it isn't repeated, I'll remove it for now.

stephenmathieson replied:
CSS is crazy 🤯

scurker commented on packages/styles/icon-button.css:
```diff
@@ -1,42 +1,42 @@
-.IconButton svg {
-  height: calc(var(--button-thin-height) - 4px);
-  width: calc(var(--button-thin-height) - 4px);
+.IconButton div {
```
One more small thing I noticed, do you think this should be more specific? i.e. `.IconButton .Icon`? 

`div` seems a bit too generic.

okry replied:
Fixed
